DYNAMIC.md

```diff
@@ -0,0 +1,326 @@
+    void setBoolean(String fieldName, boolean value);
+    void setShort(String fieldName, short value);
+    void setInt(String fieldName, int value);
+    void setLong(String fieldName, long value);
+    void setByte(String fieldName, byte value);
+    void setFloat(String fieldName, float value);
+    void setDouble(String fieldName, double value);
+    void setString(String fieldName, String value);
+    void setBlob(String fieldName, byte[] value);
+    void setDate(String fieldName, Date value);
+    void setObject(String fieldName, DynamicRealmObject value);
+    void setList(String fieldName, RealmList<DynamicRealmObjet> list);
+    boolean isNull(String fieldName);
+    boolean hasField(String fieldName);
+    String[] getFieldNames();
+    void removeFromRealm();
+}
+
+*RealmSchema*
+
+public class RealmSchema {
+    RealmObjectSchema getClass(String className);
+    RealmObjectSchema addClass(String className);
+    void removeClass(String className);
+    RealmObjectSchema renameClass(String oldName, String newName);
+}
+
+*RealmObjectSchema*
+
+public interface RealmObjectSchemaInterface {
+    RealmObjectSchema addStringField(String fieldName);
+    RealmObjectSchema addStringField(String fieldName, Set<RealmModifier> modifiers);
+    RealmObjectSchema addShortField(String fieldName);
+    RealmObjectSchema addShortField(String fieldName, Set<RealmModifier> modifiers);
+    RealmObjectSchema addIntField(String fieldName);
+    RealmObjectSchema addIntField(String fieldName, Set<RealmModifier> modifiers);
+    RealmObjectSchema addLongField(String fieldName);
+    RealmObjectSchema addLongField(String fieldName, Set<RealmModifier> modifiers);
+    RealmObjectSchema addBooleanField(String fieldName);
+    RealmObjectSchema addBooleanField(String fieldName, Set<RealmModifier> modifiers);
+    RealmObjectSchema addByteArrayField(String fieldName);
+    RealmObjectSchema addByteArrayField(String fieldName, Set<RealmModifier> modifiers);
+    RealmObjectSchema addFloatField(String fieldName);
+    RealmObjectSchema addFloatField(String fieldName, Set<RealmModifier> modifiers);
+    RealmObjectSchema addDoubleField(String fieldName);
+    RealmObjectSchema addDoubleField(String fieldName, Set<RealmModifier> modifiers);
+    RealmObjectSchema addDateField(String fieldName);
+    RealmObjectSchema addDateField(String fieldName, Set<RealmModifier> modifiers);
+    RealmObjectSchema addObjectField(String fieldName, RealmObjectSchema objectSchema);
+    RealmObjectSchema addListField(String fieldName, RealmObjectSchema objectSchema);
+    RealmObjectSchema removeField(String fieldName);
+    RealmObjectSchema renameField(String currentFieldName, String newFieldName);
+    RealmObjectSchema addIndex(String fieldName);
+    RealmObjectSchema removeIndex(String fieldName);
+    RealmObjectSchema addPrimaryKey(String fieldName);
+    RealmObjectSchema setNonNullable(String fieldName); 
+    RealmObjectSchema setNullable(String fieldName);
```

Comment:
I feel `setNullable(String fieldName, boolean nullable)` would be better. Or corresponding th `addIndex`  and `removeIndex`, maybe it should be `enableNullable` or `disableNullable`?